main.py

```diff
@@ -25,19 +30,28 @@
     boxes = face_recognition.face_locations(frame)
     encodings = face_recognition.face_encodings(frame, boxes)
     matches = []
-    for e,encoding in enumerate(encodings):
-        matches = face_recognition.compare_faces(Knownencodings,encoding)
+    foundnames = []
+    for x,encoding in enumerate(encodings):
+        matches = face_recognition.compare_faces(Knownencodings,encoding, tolerance = TOLERANCE)
         print(matches)
-        print(e)
-
-    for e,i in enumerate(matches):
-        if i:
-            print("BOXES",boxes)
-            top,right,bottom,left = boxes[0]
+        if not matches:
+            for i in boxes:
+                matches.append(False)
+        for e,i in enumerate(matches):
+            print("BOXES:",boxes)
+            top,right,bottom,left = boxes[x]
             cv2.rectangle(frame, (left,top), (right,bottom) , (0,255,0), 4)
-            image = cv2.putText(frame, names[e], (left-20,top-20), FONT, FONTSCALE,
-            TEXTCOLOUR, 1, cv2.LINE_AA, False)
 
+            if i:
+                foundnames.append(names[e])
+                image = cv2.putText(frame, names[e], (left-20,top-20), FONT, FONTSCALE,
+                TEXTCOLOUR, 1, cv2.LINE_AA, False)
+            elif True not in matches:
+                image = cv2.putText(frame, "Unknown", (left-20,top-20), FONT, FONTSCALE,
+                TEXTCOLOUR, 1, cv2.LINE_AA, False)
+
+
+    print("Found:",foundnames)
     cv2.imshow("window",frame)
     if cv2.waitKey(1) == ord("q"):
         break
```

Comment:
A good habit to get into at the end of loops like this that span the entire life of the program, is to add a 1 millisecond sleep so that the processor isn't burned out at potentially infinite speed.